src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -361,3 +429,60 @@ def generate_grant_sql_ddl_statements(
         grant_sql_statements.append(grant_sql_statement)
 
     return "\n".join(grant_sql_statements)
+
+
+def generate_new_sql_file_name(
+    deploy_root: Path, generated_root: Path, py_file: Path
+) -> Path:
+    """
+    Generates a SQL filename for the generated root from the python file, and creates its parent directories.
+    """
+    relative_py_file = py_file.relative_to(deploy_root)
+    sql_file = Path(generated_root, relative_py_file.with_suffix(".sql"))
+    if sql_file.exists():
+        cc.warning(
+            f"""\
+            File {sql_file} already exists, will append SQL statements to this file.
+        """
+        )
+    sql_file.parent.mkdir(exist_ok=True, parents=True)
+    return sql_file
+
+
+def edit_setup_script_with_exec_imm_sql(
+    collected_sql_files: List[Path], deploy_root: Path, generated_root: Path
+):
+    """
+    Adds an 'execute immediate' to setup script for every SQL file in the map
+    """
+    # Create a __generated.sql in the __generated folder
+    generated_file_path = Path(generated_root, f"{generated_root.stem}.sql")
+    generated_file_path.parent.mkdir(exist_ok=True, parents=True)
+
+    if generated_file_path.exists():
+        cc.warning(
+            "Could not complete code generation of Snowpark Extension Functions."
+        )
+        return
+
+    # For every SQL file, add SQL statement 'execute immediate' to __generated.sql script.
+    for sql_file in collected_sql_files:
+        sql_file_relative_path = sql_file.relative_to(
+            deploy_root
+        )  # Path on stage, without the leading slash
+        with open(generated_file_path, "a") as file:
+            file.write(f"\nEXECUTE IMMEDIATE FROM '/{sql_file_relative_path}';")
```

Comment:
flip this around, you should not keep opening and closing the same file to append, it's expensive. You should open the file, write in multiple batches, then close.